lang/java/avro/src/test/java/org/apache/avro/generic/TestGenericDatumWriter.java

```diff
@@ -257,4 +256,79 @@ public void writeDoesNotAllowJavaEnumForGenericEnum() throws IOException {
 
     writer.write(record, encoder);
   }
+
+  @Test
+  public void testExternalSchemaWithExplicitNullDefault() throws Exception {
+    Schema schema = createExternalSchemaWithExplicitNullDefault();
+    GenericRecord record = createRecord(schema);
+    writeObject(schema, record);
+  }
+
+  @Test
+  public void testExternalSchemaWithoutExplicitNullDefault() throws Exception {
+    Schema schema = createExternalSchemaWithoutExplicitNullDefault();
+    GenericRecord record = createRecord(schema);
+    writeObject(schema, record);
+  }
+
+  @Test
+  public void testCodeGeneratedSchemaWithoutExplicitNullDefault() throws Exception {
```

Comment:
I don't think both the testExternal and testCodeGenerated tests are needed as they only differ in how the schema is created.